ui/pages/confirmations/hooks/alerts/signatures/useDomainMismatchAlert.ts

```diff
@@ -0,0 +1,46 @@
+import { useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import {
+  isValidSIWEOrigin,
+  WrappedSIWERequest,
+} from '@metamask/controller-utils';
+
+import { Alert } from '../../../../../ducks/confirm-alerts/confirm-alerts';
+import { Severity } from '../../../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../../../hooks/useI18nContext';
+import { currentConfirmationSelector } from '../../../../../selectors';
+
+import { SignatureRequestType } from '../../../types/confirm';
+import { isSIWESignatureRequest } from '../../../utils';
+
+export default function useDomainMismatchAlerts(): Alert[] {
+  const t = useI18nContext();
+
+  const currentConfirmation = useSelector(
+    currentConfirmationSelector,
+  ) as SignatureRequestType;
+  const isSIWE = isSIWESignatureRequest(currentConfirmation);
+
+  if (!isSIWE || !currentConfirmation?.msgParams) {
+    return [];
```

Comment:
Since we're using a `useMemo` below, we can't early return as we need to call all the hooks on every render.

But we can instead fold these conditions into our `isValidDomain` boolean.

Reply:
updated [3763396](https://github.com/MetaMask/metamask-extension/pull/25281/commits/376339668fc6b1f2ca792d6e121b2e57c35e1848)